File: packages/events/src/catalog.ts
export const EVENT_CATALOG = {
  booking: {
    source: 'tc.booking' as const,
    events: {
      BookingCreated: 'BookingCreated',
      BookingAccepted: 'BookingAccepted',
      BookingCancelled: 'BookingCancelled',
      BookingCompleted: 'BookingCompleted',
      BookingStatusChanged: 'BookingStatusChanged',
    },
  },
  matching: {
    source: 'tc.matching' as const,
    events: {
      ProviderMatched: 'ProviderMatched',
      MatchTimeout: 'MatchTimeout',
      MatchReassigned: 'MatchReassigned',
    },
  },
  tracking: {
    source: 'tc.tracking' as const,
    events: {
      LocationUpdated: 'LocationUpdated',
      DriverArrived: 'DriverArrived',
      RouteDeviation: 'RouteDeviation',
    },
  },
  payment: {
    source: 'tc.payment' as const,
    events: {
      PaymentInitiated: 'PaymentInitiated',
      PaymentCompleted: 'PaymentCompleted',
      PaymentFailed: 'PaymentFailed',
      RefundProcessed: 'RefundProcessed',
    },
  },
  sos: {
    source: 'tc.sos' as const,
    events: {
      SOSActivated: 'SOSActivated',
      SOSResolved: 'SOSResolved',
    },
  },
  auth: {
    source: 'tc.auth' as const,
    events: {
      UserRegistered: 'UserRegistered',
      UserSuspended: 'UserSuspended',
    },
  },
  provider: {
    source: 'tc.provider' as const,
    events: {
      ProviderOnline: 'ProviderOnline',
      ProviderOffline: 'ProviderOffline',
      ProviderVerified: 'ProviderVerified',
    },
  },
  evidence: {
    source: 'tc.evidence' as const,
    events: {
      EvidenceUploaded: 'EvidenceUploaded',
      ReportCompleted: 'ReportCompleted',
    },
  },
} as const;

File: internal/domain/event/catalog.go
// Package event defines domain event sources and detail types for the TowCommand platform.
package event

// Event sources identify the bounded context that originated the event.
const (
	SourceBooking  = "tc.booking"
	SourceMatching = "tc.matching"
	SourceTracking = "tc.tracking"
	SourcePayment  = "tc.payment"
	SourceSOS      = "tc.sos"
	SourceAuth     = "tc.auth"
	SourceEvidence = "tc.evidence"
)

// Booking detail types.
const (
	BookingCreated       = "BookingCreated"
	BookingMatched       = "BookingMatched"
	BookingCancelled     = "BookingCancelled"
	BookingStatusChanged = "BookingStatusChanged"
	BookingCompleted     = "BookingCompleted"
)

// Matching detail types.
const (
	MatchingStarted   = "MatchingStarted"
	MatchingCompleted = "MatchingCompleted"
	MatchingFailed    = "MatchingFailed"
	ProviderAccepted  = "ProviderAccepted"
	ProviderDeclined  = "ProviderDeclined"
)

// Tracking detail types.
const (
	LocationUpdated = "LocationUpdated"
	ETAUpdated      = "ETAUpdated"
	GeofenceEntered = "GeofenceEntered"
	GeofenceExited  = "GeofenceExited"
)

// Payment detail types.
const (
	PaymentInitiated = "PaymentInitiated"
	PaymentCaptured  = "PaymentCaptured"
	PaymentRefunded  = "PaymentRefunded"
	PaymentFailed    = "PaymentFailed"
)

// SOS detail types.
const (
	SOSTriggered   = "SOSTriggered"
	SOSResolved    = "SOSResolved"
	RouteDeviation = "RouteDeviation"
)

// Auth detail types.
const (
	UserRegistered    = "UserRegistered"
	UserVerified      = "UserVerified"
	ProviderOnboarded = "ProviderOnboarded"
)

// Provider detail types.
const (
	SourceProvider      = "tc.provider"
	ProviderRegistered  = "ProviderRegistered"
	ProviderOnline      = "ProviderOnline"
	ProviderOffline     = "ProviderOffline"
	AvailabilityChanged = "AvailabilityChanged"
)

// Evidence detail types.
const (
	EvidenceUploaded       = "EvidenceUploaded"
	ConditionReportCreated = "ConditionReportCreated"
	EvidenceValidated      = "EvidenceValidated"
)

// Rating detail types.
const (
	SourceRating    = "tc.rating"
	RatingSubmitted = "RatingSubmitted"
)

// OTP detail types.
const (
	SourceOTP    = "tc.otp"
	OTPGenerated = "OTPGenerated"
	OTPVerified  = "OTPVerified"
)
